### DISCUSSION TITLE
cluster origin-edge

### DISCUSSION THREAD

**Discussion by NkInKaIgIe** [OvenMediaLabs/OvenMediaEngine#1233] [Help] [answered]
i have origin-edge cluster. when stream has streamkey <abcxyz> for origin 1 domain with multiple ip vm. then 1 streamkey can be streamed to many vm . Is there any way to solve this problem?. What origin will Edge get the stream from?
example:
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/96049216/50ddf7e1-de74-4337-bf4e-a8ef85bbdca3)

2 streams with streamkey 1=abcxyz and streamkey 2=abcxyz
then when starting live, Origin 1 will have a stream with streamkey1=abcxyz and Origin 2 will have a stream with streamkey2 =abcxyz (origin use the same domain)

**Comment by getroot** [comment, marked as answer]
If you use OriginMap, you can map the playback URL to a Location and point it to the origin server of your choice.

If you use OriginMapStore, the app/stream name must be unique across the entire cluster.

https://airensoft.gitbook.io/ovenmediaengine/origin-edge-clustering

**Comment by NkInKaIgIe** [comment]
@getroot   input obs 2 general live streamkey. streamkey=abcxyz :
2 obs live stream rtmp://origin.live:1935/app/abcxyz , then the origin in the cluster will keep 2 streams with the same streamkey in each origin .
Input using load balancer will select the origin. How to directly connect 2 streams streamkey=abcxyz to the cluster allows only 1 stream to work on the whole cluster.
how to fix 1 streamkey has only 1 stream ?

**Comment by NkInKaIgIe** [comment]
@getroot  thank you very much . I used OriginMapStore and redis solved this problem
